server/bound_queries.cc

```diff
@@ -238,7 +238,7 @@ struct UnorderedExecutor: public BoundSelectQuery::Executor {
         // Do we select *?  In that case we can avoid a lot of copying
         bool selectStar = boundSelect.expr->isIdentitySelect(context);
 
-        int numRows = whereGenerator.upperBound;
+        int64_t numRows = whereGenerator.rowStreamTotalRows;
```

Comment:
No.  We use -1 as a guard value, which we don't want to become MAXINT if we don't check the value properly.